app/src/hooks/useKeycloakWrapper.tsx

```diff
@@ -261,41 +197,36 @@ function useKeycloakWrapper(): IKeycloakWrapper {
   };
 
   const username = (): string | undefined => {
-    return keycloakUser?.preferred_username;
+    return keycloakUserDataLoader.data?.preferred_username;
   };
 
   const displayName = (): string | undefined => {
-    return keycloakUser?.name || keycloakUser?.preferred_username;
+    return keycloakUserDataLoader.data?.name || keycloakUserDataLoader.data?.preferred_username;
   };
 
   const email = (): string | undefined => {
-    return keycloakUser?.email;
+    return keycloakUserDataLoader.data?.email;
   };
 
   const firstName = (): string | undefined => {
-    return keycloakUser?.firstName;
+    return keycloakUserDataLoader.data?.firstName;
   };
 
   const lastName = (): string | undefined => {
-    return keycloakUser?.lastName;
+    return keycloakUserDataLoader.data?.lastName;
   };
 
   const refresh = () => {
-    // refresh system user
-    setHasLoadedSystemUser(false);
-    setIsSystemUserLoading(false);
-    setShouldLoadSystemUser(true);
-
-    // refresh access requests
-    setShouldLoadAccessRequest(true);
+    userDataLoader.refresh();
+    hasPendingAdministrativeActivitiesDataLoader.refresh();
   };
 
   return {
     keycloak: keycloak,
-    hasLoadedAllUserInfo: hasLoadedSystemUser,
+    hasLoadedAllUserInfo: !!(userDataLoader.data || hasPendingAdministrativeActivitiesDataLoader.data),
     systemRoles: getSystemRoles(),
     hasSystemRole,
-    hasAccessRequest,
+    hasAccessRequest: !!hasPendingAdministrativeActivitiesDataLoader.data,
```

Comment:
Thoughts on the `Boolean` constructor vs. the `!!` operator?

Reply:
Hadn't considered Boolean tbh.  Did a quick search and it seems like both are totally fine, no relevant performance difference or anything. Just comes down to preference, and maybe landing on one of them for consistency.

Some people liked Boolean because its easier to read, others liked !! because its more concise.

The airbnb style guide is possibly the most popular one, so when in doubt not a bad place to check: https://airbnb.io/javascript/